src/components/BarChart/WalletChart.tsx

```diff
@@ -1,34 +1,53 @@
-import React, { useMemo, useState } from 'react'
-import { TouchableWithoutFeedback } from 'react-native'
-import { random, times } from 'lodash'
+import React, { useEffect, useState, useMemo, useCallback } from 'react'
+import { ActivityIndicator, TouchableWithoutFeedback } from 'react-native'
+import { round } from 'lodash'
+import { useSelector } from 'react-redux'
 import ChartContainer from './ChartContainer'
 import CarotLeft from '../../assets/images/carot-left.svg'
 import CarotRight from '../../assets/images/carot-right.svg'
 import Box from '../Box'
 import Text from '../Text'
-import { ChartData } from './types'
+import { ChartData, ChartRange } from './types'
 import { triggerImpact } from '../../utils/haptic'
 import { useColors } from '../../theme/themeHooks'
+import { useAppDispatch } from '../../store/store'
+import { RootState } from '../../store/rootReducer'
+import { fetchActivityChart } from '../../store/account/accountSlice'
 
 type Props = {
   height: number
 }
 
 const WalletChart = ({ height }: Props) => {
+  const dispatch = useAppDispatch()
+  const {
+    account: { activityChart },
+    activity: { filter },
+  } = useSelector((state: RootState) => state)
+
   const [focusedData, setFocusedData] = useState<ChartData | null>(null)
-  const [timeframe, setTimeframe] = useState(0)
+  const [timeframe, setTimeframe] = useState<ChartRange>('daily')
+
+  const data = useMemo(() => activityChart[timeframe].data, [
+    activityChart,
+    timeframe,
+  ])
+
+  useEffect(() => {
+    dispatch(fetchActivityChart({ range: timeframe, filterType: filter }))
+  }, [dispatch, timeframe, filter])
 
   const headerHeight = 30
   const padding = 20
-  const chartHeight = height - headerHeight - padding
+  const chartHeight = useMemo(() => height - headerHeight - padding, [height])
 
-  const changeTimeframe = (t: number) => {
-    setTimeframe(t)
+  const changeTimeframe = useCallback((range: ChartRange) => {
```

Comment:
you could curry this function so you don't have to have inline functions
```
  const changeTimeframe = useCallback(
    (range: ChartRange) => () => {
      setTimeframe(range)
      triggerImpact()
    },
    [],
  )
```

Then just call it right away...
```
 <TouchableWithoutFeedback onPress={changeTimeframe('daily')}>
 ```